jooqdb/src/main/java/org/cristalise/storage/jooqdb/JooqClusterStorage.java

```diff
@@ -1,339 +1,426 @@
+        jooqHandlers.put(ClusterType.LIFECYCLE,  new JooqLifecycleHandler());
+        jooqHandlers.put(ClusterType.COLLECTION, new JooqCollectionHadler());
+        jooqHandlers.put(ClusterType.HISTORY,    new JooqHistoryHandler());
+        jooqHandlers.put(ClusterType.JOB,        new JooqJobHandler());
+        jooqHandlers.put(ClusterType.ATTACHMENT, new JooqOutcomeAttachmentHandler());
+
+        DSLContext context = JooqHandler.connect();
+
+        context.transaction(nested -> {
+            for (JooqHandler handler: jooqHandlers.values()) handler.createTables(DSL.using(nested));
+        });
+
+        try {
+            String handlers = Gateway.getProperties().getString(JOOQ_DOMAIN_HANDLERS, "");
+
+            for(String handlerClass: StringUtils.split(handlers, ",")) {
+                if (!handlerClass.contains(".")) handlerClass = "org.cristalise.storage."+handlerClass;
+
+                Logger.msg(1, "JooqClusterStorage.initialiseHandlers() - Instantiate domain handler:"+handlerClass);
+
+                domainHandlers.add( (JooqDomainHandler) Class.forName(handlerClass).newInstance());
+            }
+        }
+        catch (InstantiationException | IllegalAccessException | ClassNotFoundException ex) {
+            Logger.error("JooqClusterStorage could not instantiate domain handler");
+            Logger.error(ex);
+            throw new PersistencyException("JooqClusterStorage could not instantiate domain handler:"+ex.getMessage());
+        }
+
+        if (! Gateway.getProperties().getBoolean(JOOQ_DISABLE_DOMAIN_CREATE, false)) {
+            context.transaction(nested -> {
+                for (JooqDomainHandler handler: domainHandlers) handler.createTables(DSL.using(nested));
+            });
+        }
+    }
+
+    public void dropHandlers() throws PersistencyException {
+        DSLContext context = JooqHandler.connect();
+        for (JooqHandler handler: jooqHandlers.values()) handler.dropTables(context);
+    }
+
+    @Override
+    public void close() throws PersistencyException {
+        Logger.msg(1, "JooqClusterStorage.close()");
+        try {
+            JooqHandler.connect().close();
+        }
+        catch (Exception e) {
+            Logger.error(e);
+            throw new PersistencyException(e.getMessage());
+        }
+    }
+
+    @Override
+    public void postBoostrap() throws PersistencyException {
+        JooqHandler.connect().transaction(nested ->{
+            for (JooqDomainHandler domainHandler : domainHandlers) domainHandler.postBoostrap(DSL.using(nested));
+        });
+        
+        if(Gateway.getProperties().containsKey(JOOQ_AUTOCOMMIT)){
+
+            Gateway.getProperties().put(JOOQ_AUTOCOMMIT, false);
+            HikariConfig config = new HikariConfig();
+            JooqHandler.ds.copyStateTo(config);
+            config.setAutoCommit(false);
+            config.addDataSourceProperty( "autoCommit",false);

```

Comment:
should this be also changed to `true` later, like `JOOQ_AUTOCOMMIT` is changed in `postConnect()`?

Reply:
Nope. This should set to false and will be used all throughout after the bootstrap.